src/hooks/useTabs.ts

```diff
@@ -0,0 +1,15 @@
+import TabsBase from "../Explorer/Tabs/TabsBase";
+import { TabsManager } from "../Explorer/Tabs/TabsManager";
+import { useObservableState } from "./useObservableState";
+
+export type UseTabs = {
+  tabs: readonly TabsBase[];
+  tabsManager: TabsManager;
+};
+
+export function useTabs(): UseTabs {
+  const tabsManager = new TabsManager();
```

Comment:
Does this create a new tabs manager instance on every render and if so is that an issue?

`useState(() => newTabsManager())` would ensure you get only one instance

Reply:
Yes... but so far only `<App>` uses this. I'm guessing that if `<App>`  is being re-rendered it's because we're restarting the whole thing?